File: greedy/LeetCode55.go
package greedy

func max(x, y int) int {
	if x < y {
		return y
	}
	return x
}

func canJump(nums []int) bool {
	if len(nums) == 0 {
		return false
	}
	maxJump := 0
	for i := 0; i < len(nums); i++ {
		if maxJump < i {
			return false
		}
		maxJump = max(i+nums[i], maxJump)
	}
	return true
}

File: greedy/LeetCode55.cc
#include<vector>
#include<algorithm>
using namespace std;

class Solution {
public:
    // bool canJump(vector<int>& nums) {
    //     sort(nums.begin(),nums.end());
    //     if(nums[0] == 0 && nums.size() > 0){
    //         return false;
    //     }
    //     for(int i = 0; i < nums.size(); i++){
    //         if(i += nums[i] >= nums.size()){
    //             return true;
    //         }
    //     }
    // }

    bool canJump(vector<int>& nums) {
        int ans = 0;
        for(int i = 0; i < nums.size(); i++){
            if(i <= ans){
                ans = max(ans,i+nums[i]);
                if(ans >= nums.size()-1){
                    return true;
                }
            }
        }
        return false;
    }
};